scrapy/downloadermiddlewares/httpcompression.py

```diff
@@ -78,9 +81,35 @@ def open_spider(self, spider: Spider) -> None:
         if hasattr(spider, "download_warnsize"):
             self._warn_size = spider.download_warnsize
 
+    @staticmethod
+    def _get_unsupported_suggestion(encoding: bytes) -> str:
+        if encoding == b"br":
+            return " You need to install brotli or brotlicffi to use it."
+        if encoding == b"zstd":
+            return " You need to install zstandard to use it."
+        return ""
+
     def process_request(
         self, request: Request, spider: Spider
     ) -> Request | Response | None:
+
+        user_header = request.headers.get("Accept-Encoding")
+        if user_header:
+            for user_encoding in user_header.split(b","):
+                user_encoding = user_encoding.strip()
+                # Encoding specification can also include a weight,
+                # we won't check such encodings.
```

Comment:
Would this do the job?

```suggestion
                user_encoding = user_encoding.split(";")[0].strip()
```

Reply:
What if the weight is 0? :)

I just don't know if there is a need to complicate the code here.

Reply:
Oh, I did not realize that was a possibility! :+1: